File: task-3/src/main/java/ru/biluta/MainLauncher.java
package ru.biluta;

import jade.core.Profile;
import jade.core.ProfileImpl;
import jade.core.Runtime;
import jade.wrapper.AgentContainer;
import jade.wrapper.AgentController;
import jade.wrapper.StaleProxyException;
import ru.biluta.agents.AgentCompany;
import ru.biluta.agents.AgentManufacturer;

public class MainLauncher {
    public static void main(String[] args) {
        Runtime rt = Runtime.instance();
        Profile profile = new ProfileImpl();
        profile.setParameter(Profile.MAIN_HOST, "localhost");
//        profile.setParameter(Profile.GUI, "true");

        AgentContainer ac = rt.createMainContainer(profile);

        try {
            AgentController manufacturer1 = ac.createNewAgent(
                    "manufacturer1",
                    AgentManufacturer.class.getName(),
                    new Object[]{"Фотон", 120, 9, 5}
            );
            AgentController manufacturer2 = ac.createNewAgent(
                    "manufacturer2",
                    AgentManufacturer.class.getName(),
                    new Object[]{"Спецдеталь", 200, 7, 8}
            );
            AgentController manufacturer3 = ac.createNewAgent(
                    "manufacturer3",
                    AgentManufacturer.class.getName(),
                    new Object[]{"Станкострой", 100, 8, 5}
            );
            AgentController manufacturer4 = ac.createNewAgent(
                    "manufacturer4",
                    AgentManufacturer.class.getName(),
                    new Object[]{"Атом", 150, 6, 7}
            );
            AgentController company1 = ac.createNewAgent(
                    "company1",
                    AgentCompany.class.getName(),
                    new Object[]{"Фирма1", 100, 8, 5}
            );
            AgentController company2 = ac.createNewAgent(
                    "company2",
                    AgentCompany.class.getName(),
                    new Object[]{"Фирма2", 150, 5, 6}
            );

            manufacturer1.start();
            manufacturer2.start();
            manufacturer3.start();
            manufacturer4.start();
            company1.start();
            company2.start();
        } catch (StaleProxyException e) {
            System.out.println("Exception starting agent: " + e.getMessage());
            e.printStackTrace();
        }
    }
}

File: task-4-new2/src/main/kotlin/MainLauncher.kt
package ru.biluta

import jade.core.Profile
import jade.core.ProfileImpl
import jade.core.Runtime
import jade.wrapper.AgentContainer
import jade.wrapper.AgentController
import ru.biluta.agents.OrderGeneratorAgent
import ru.biluta.agents.WarehouseAgent
import ru.biluta.agents.WarehouseManagerAgent
import ru.biluta.agents.WorkerAgent
import kotlin.random.Random

fun main() {
    val runtime: Runtime = Runtime.instance()
    val profile: Profile = ProfileImpl()
    val mainContainer: AgentContainer = runtime.createMainContainer(profile)

    val inventory = OrderUtils.items.associateWith { item ->
        //val quantity = Random.nextInt(0, 30)
        val quantity = 100
        quantity
    }
    val inventoryArgs = arrayOf(inventory)

    val numOrders = 5
    val orderAgentArgs = arrayOf(numOrders)

    try {

        val warehouseAgent: AgentController = mainContainer.createNewAgent("warehouse", WarehouseAgent::class.java.name, inventoryArgs)
        warehouseAgent.start()

        val managerAgent: AgentController = mainContainer.createNewAgent("warehouseManager", WarehouseManagerAgent::class.java.name, null)
        managerAgent.start()

        for (i in 1..3) {
            val workerAgent: AgentController = mainContainer.createNewAgent("worker$i", WorkerAgent::class.java.name, null)
            workerAgent.start()
        }

        val orderGeneratorAgent: AgentController = mainContainer.createNewAgent("orderGenerator", OrderGeneratorAgent::class.java.name, orderAgentArgs)
        orderGeneratorAgent.start()



    } catch (e: Exception) {
        e.printStackTrace()
    }
}